### PR TITLE
[BACKLOG-38398] Allow users to specify the names of specific pairs of

### PR DESCRIPTION
**Pull request by peterrinehart**
date parameters that will provide the option to be set using dynamically-calculated values based on a time interval relative to the time the report is run.  Checks the classic-engine.properties file for param names and injects additional parameters when the report is scheduled to allow selecting a relative date range.

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 14m
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **539**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-804/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/reporting/platform/plugin/ReportContentUtil.java [hunk 1]
```diff
@@ -102,6 +121,103 @@ public static ValidationResult applyInputsToReportParameters( final MasterReport
...
+
+  private static String getRelativeValParamName( String paramName ) {
+    return paramName + "_RelativeVal";
+  }
+
+  private static String getRelativeUnitParamName( String paramName ) {
+    return paramName + "_RelativeUnit";
+  }
+
+  protected static boolean injectRelativeDateParams( ParameterDefinitionEntry[] inputParams ) {
```
[the remark this answers is not in the document]
Comment (peterrinehart) [reply core/src/main/java/org/pentaho/reporting/platform/plugin/ReportContentUtil.java:140]: I can change that, sure.

File: core/src/main/java/org/pentaho/reporting/platform/plugin/ReportContentUtil.java [hunk 2]
```diff
@@ -102,6 +121,103 @@ public static ValidationResult applyInputsToReportParameters( final MasterReport
...
+    unitParam.addValues( "Days", "Days" );
+    unitParam.addValues( "Weeks", "Weeks" );
+    unitParam.addValues( "Months", "Months" );
+    unitParam.addValues( "Months (Calendar)", "Months (Calendar)" );
+    unitParam.addValues( "Quarter (Calendar)", "Quarter (Calendar)" );
+    unitParam.addValues( "Quarter (Fiscal)", "Quarter (Fiscal)" );
+    unitParam.addValues( "Months", "Months" );
+    unitParam.addValues( "Years", "Years" );
+    unitParam.addValues( "Years (Calendar)", "Years (Calendar)" );
+    unitParam.addValues( "Years (Fiscal)", "Years (Fiscal)" );
```
[the remark this answers is not in the document]
Comment (peterrinehart) [reply core/src/main/java/org/pentaho/reporting/platform/plugin/ReportContentUtil.java:196]: Write this up as another issue please.  Should be doable but might take a couple of days to sort out.

[the remark this answers is not in the document]
Comment (pentaho-whartman) [reply core/src/main/java/org/pentaho/reporting/platform/plugin/ReportContentUtil.java:196]: ticket here: https://hv-eng.atlassian.net/browse/BACKLOG-38671

**Review by pentaho-whartman — APPROVED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 13m 10s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **539**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-804/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**